core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderService.js

```diff
@@ -0,0 +1,158 @@
+      var htmlString;
+      if (currentStateName === destStateName) {
+        htmlString =
+          'Submitted answer "' + answer + '" in card "' + currentStateName +
+          '".<br><br>';
+      } else {
+        htmlString =
+          'Submitted answer "' + answer + '" and moved to card "' +
+          destStateName + '" after spending ' + timeSpentInStateSecs +
+          ' seconds on card "' + currentStateName + '".<br><br>';
+      }
+      return htmlString;
+    };
+
+    var renderLearnerActionsTableForMultipleIncorrectIssue = function(
+        finalBlock) {
+      var index = finalBlock.length - 1;
+      var stateName =
+        finalBlock[index].actionCustomizationArgs.state_name.value;
+
+      var tableHTML = '<table><tr><th>Answer</th><th>Feedback</th></tr>';
+      for (var i = 0; i < index; i++) {
+        if (finalBlock[i].actionType !== ACTION_TYPE_ANSWER_SUBMIT) {
+          continue;
+        }
+        var answer = finalBlock[i].actionCustomizationArgs.answer.value;
+        var feedback = finalBlock[i].actionCustomizationArgs.feedback.value;
+        tableHTML +=
+          '<tr><td>' + answer + '</td><td>' + feedback + '</td></tr>';
+      }
+      tableHTML += '</table>';
+      return tableHTML;
+    };
+
+    var renderLearnerActionHTML = function(learnerAction) {
+      var actionType = learnerAction.actionType;
+      var custArgs = learnerAction.actionCustomizationArgs;
+      if (actionType === ACTION_TYPE_EXPLORATION_START) {
+        return renderExplorationStartActionHTML(custArgs.state_name.value);
+      } else if (actionType === ACTION_TYPE_EXPLORATION_QUIT) {
+        return renderExplorationQuitActionHTML(
+          custArgs.state_name.value, custArgs.time_spent_in_state_secs.value);
+      } else if (actionType === ACTION_TYPE_ANSWER_SUBMIT) {
+        interactionId =
+          custArgs.interaction_id.value;
+        if (interactionId === 'Continue') {
+          return renderContinueButtonSubmitActionHTML(
+            custArgs.dest_state_name.value,
+            custArgs.time_spent_in_state_secs.value);
+        }
+        if (custArgs.answer.value) {
+          return renderAnswerSubmitActionHTML(
+            custArgs.answer.value, custArgs.dest_state_name.value,
+            custArgs.time_spent_in_state_secs.value, custArgs.state_name.value
+          );
+        }
+      }
+    };
+
+    return {
+      renderFinalDisplayBlockForMISIssueHTML: function(block) {
+        var index = block.length - 1;
+        var stateName = block[index].actionCustomizationArgs.state_name.value;
+        var htmlString = '';
+        for (
+          var i = 0; block[i].actionType !== ACTION_TYPE_ANSWER_SUBMIT; i++) {
+          htmlString += renderLearnerActionHTML(block[i]);
+        }
+        htmlString +=
+          'Submitted the following answers in card "' + stateName + '"<br><br>';
+        htmlString += renderLearnerActionsTableForMultipleIncorrectIssue(block);
+        htmlString += '<br><br>';
```

Comment:
Again, this sort of spacing should be handled by CSS, not arbitrary tag blocks. Please use a `<span class="something">`, so that we can change the `margin-bottom` of it later.

Reply:
Done.